### ISSUE TITLE
Move config from Rails to SDoc

### ISSUE THREAD

**Issue by p8** [rails/sdoc#333]
Currently generating the API docs is integrated in the Rails project:
https://github.com/rails/rails/blob/main/railties/lib/rails/api/task.rb
This requires the sdoc gem to work with the supported Ruby version of the Rails project and its Gemfile.
If docs generation can be run seperately we can generate the docs for any version of Rails.

**Comment by zzak** [comment]
Yeah, this is a known issue but I'm not sure where to put it yet.

**Comment by p8** [comment]
I think we could have some config file in sdoc where we define the settings for each version of Rails.

**Comment by p8** [comment]
RDoc [supports ](https://ruby.github.io/rdoc/RDoc/Options.html#class-RDoc::Options-label-Saved+Options)a .rdoc_options file
We could move a lot of options there.

**Comment by zzak** [comment]
Ok, the last issue. I left this because I think it's actually nice backend work we can do without disrupting any UX stuff.

The goal is to have SDoc be able to generate the Rails API docs for any version. So not relying on configuration based in-repo.

This way we can have a single process to build the docs for all (stable) versions of Rails when we deploy, and changes or improvements to the UI will be applied everywhere. That removes the jarring effect of viewing older version of the docs and getting lost (not that I don't love the nostalgia of the old ui).
